core/lib/src/response/responder.rs

```diff
@@ -217,34 +219,156 @@ impl<'r> Responder<'r> for String {
     }
 }
 
+fn range_response<'r, B: io::Read + io::Seek + 'r>(range: Range, mut body: B) -> Result<response::Response<'r>, B> {
+    use std::cmp;
+    use http::hyper::header::{ContentRange, ByteRangeSpec, ContentRangeSpec};
+
+    match range {
+        Range::Bytes(ranges) => {
+            if ranges.len() == 1 {
+                let (start, end, total) = match ranges[0] {
```

Comment:
These match arms duplicate a decent amount of code (e.g. `let size = `, calling `seek()`). I think It will be easier to see the differences between the blocks if the common code was pulled of the match arms.

Reply:
I have pulled out both seek calls.